spec.html

```diff
@@ -30522,6 +30522,7 @@
             1. Let _index_ be _O_.[[ArrayIteratorNextIndex]].
             1. Let _itemKind_ be _O_.[[ArrayIterationKind]].
             1. If _a_ has a [[TypedArrayName]] internal slot, then
+              1. Perform ? ValidateTypedArray(_a_).
```

Comment:
Nitpicky, but it seems to me like

    1. If IsDetachedBuffer(_a_.[[ViewedArrayBuffer]]) is *true*, throw a *TypeError* exception.

is clearer and more readable about intent, and it doesn't reuse an overbroad spec method.